aesara/tensor/random/basic.py

```diff
@@ -1778,7 +1786,10 @@ def __call__(self, a, size=None, replace=True, p=None, **kwargs):
             The probabilities associated with each entry in `a`. If not
             given, all elements have equal probability.
         """
-        a = as_tensor_variable(a, ndim=1)
+        a = as_tensor_variable(a)
+
+        if a.ndim == 0:
+            a = aesara.tensor.arange(a)
 
         if p is None:
             p = aesara.tensor.type_other.NoneConst.clone()
```

Comment:
I'm not sure that we should be cloning this anymore.

Reply:
What was the intent originally?

Reply:
I believe it was related to the old need to clone every `Variable`.